### PR TITLE
Checklist qa interface

### PR DESCRIPTION
**Pull request by adbharadwaj**
The interface for evaluation using the template todolist used by the experts.
- Each todo item can be approved/disapproved independent of its descendants.
- An item can be approved/disapproved without a reason. 
- To provide a reason a user can click on `add reason` link. This link appears only when the user has approved or disapproved an item.
- A user can update the reason by changing the reason text and clicking on the corresponding submit button. The submit button appears only when the reason has been updated.
- Only users in `todolist_qa_feature` can access the QA interface.

screen shot 2018-07-03 at 1 54 31 pm

The interface for showing recommendations based on QA from the latest project done by the user.
screen shot 2018-07-03 at 1 59 12 pm

### CODE REVIEW

File: orchestra/admin.py [hunk 1]
```diff
@@ -186,6 +187,13 @@ class TodoAdmin(AjaxSelectAdmin):
     list_filter = ('task__project__workflow_version',)
 
 
+@admin.register(TodoQA)
+class TodoQAAdmin(AjaxSelectAdmin):
+    list_display = ('id', 'created_at', 'todo', 'approval_reason', 'approved')
+    ordering = ('-created_at',)
+    search_fields = ('todo_description',)
```
Comment (marcua) [inline orchestra/admin.py]: todo__description (two underscores)

do you also want approval_reason (one underscore)?

File: orchestra/migrations/0078_todoqa.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
```
Comment (marcua) [inline orchestra/migrations/0078_todoqa.py:1]: you could recreate 0078 and 0079 so there's only one :)

File: orchestra/models/core/models.py [hunk 3]
```diff
@@ -659,6 +660,31 @@ class Meta:
...
+    Constraints:
+        `todo` has to be unique.
+    """
+    class Meta:
+        app_label = 'orchestra'
+
+    todo = models.OneToOneField(
+        Todo, related_name='qa', on_delete=models.CASCADE)
+    approval_reason = models.TextField(null=True, blank=True)
+    approved = models.NullBooleanField(blank=True)
```
Comment (marcua) [inline orchestra/models/core/models.py]: i feel less strongly about this, but you could make this `approved_datetime` for consistency :)

File: orchestra/models/core/models.py [hunk 4]
```diff
@@ -659,6 +660,31 @@ class Meta:
...
+
+    Constraints:
+        `todo` has to be unique.
+    """
+    class Meta:
+        app_label = 'orchestra'
+
+    todo = models.OneToOneField(
+        Todo, related_name='qa', on_delete=models.CASCADE)
+    approval_reason = models.TextField(null=True, blank=True)
```
Comment (marcua) [inline orchestra/models/core/models.py]: comment? you might leave whether you approve or disapprove

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 5]
```diff
@@ -15,6 +15,7 @@
           }"></i>
         <i class="fa fa-circle-o" aria-hidden="true" ng-if="(!todo.items || todo.items.length === 0) && showSkipped" data-nodrag></i>
         <span ng-hide="!todo.items || todo.items.length < 1 || showSkipped">[{{ ( todo.items | filter: { completed: true, skipped_datetime: '!' } ).length }}/{{ ( todo.items | filter: { skipped_datetime: '!' } ).length }}] </span>
+        <!-- <span class="todo__role" ng-if="recommendations[todo.description]">**</span> -->
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-checklist.html]: ?

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 6]
```diff
@@ -28,11 +29,14 @@
...
       <span ng-if="templates.length > 0">
         &nbsp;&nbsp;
-        <a href="#" ng-hide="!showSkipped" ng-click="unskipTodo(todo)">[add]</a>
-        <a href="#" ng-hide="showSkipped" ng-click="skipTodo(todo)">[skip]</a>
+        <a href="#" ng-hide="!showSkipped" ng-click="unskipTodo(todo)">[relevant]</a>
+        <a href="#" ng-hide="showSkipped" ng-click="skipTodo(todo)">[not relevant]</a>
         <a href="#" ng-hide="todo.parent_todo" ng-click="removeTodo(todo)">[remove]</a>
       </span>
     </label>
+    <form class="form-inline" ng-if="recommendations[todo.description] && recommendations[todo.description].approval_reason" >
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-checklist.html]: can't we use todo.id here instead of todo.description since there's a foreign key?

File: orchestra/models/core/models.py [hunk 7]
```diff
@@ -659,6 +660,31 @@ class Meta:
     activity_log = JSONField(default={'actions': []})
 
 
+class TodoQA(TodoQAMixin, BaseModel):
```
Comment (marcua) [inline orchestra/models/core/models.py:663]: Do we care who the reviewer is?

File: orchestra/static/orchestra/todos/todo-qa-list.directive.es6.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      scope.submitReason = (todo) => {
+        scope.updateTodoApprovalReason(todo)
+        scope.changedReasons[todo.id] = false
+      }
+
+      scope.addReason = (todo) => {
+        todo.qa.approval_reason = ' '
+      }
+
+      scope.isDesignTodo = (todo) => {
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-qa-list.directive.es6.js]: the word design shouldn't appear anywhere in the PR, since this is a general-purpose piece of functionality :)

File: orchestra/static/orchestra/todos/todo-qa-list.scss [hunk 9]
```diff
@@ -0,0 +1,16 @@
+.todo {
+  &-skipped {
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-qa-list.scss]: does BEM say we do -- instead of -?

(no strong feeling)

File: orchestra/static/orchestra/todos/todos.module.es6.js [hunk 10]
```diff
@@ -3,15 +3,19 @@
...
 import todoApi from 'orchestra/todos/todos.service.es6.js'
+import todoQAApi from 'orchestra/todos/todo-qas.service.es6.js'
 import todoListTemplateApi from 'orchestra/todos/todolist-template.service.es6.js'
 import 'angular-ui-tree'
 
 const name = 'orchestra.todos'
 angular.module(name, ['ui.select', 'ngSanitize', 'ui.tree', common])
   .directive('todoList', todoList)
   .directive('todoChecklist', todoChecklist)
+  .directive('todoQaList', todoQAList)
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todos.module.es6.js]: this says Qa, and the next added line says QA.

My vote is Qa :)

File: orchestra/todos/auth.py [hunk 11]
```diff
@@ -2,17 +2,25 @@
...
         worker = Worker.objects.get(user=request.user)
-        project = todo.task.project
+        if isinstance(obj, Todo):
+            project = obj.task.project
+        elif isinstance(obj, TodoQA):
+            project = obj.todo.task.project
+        else:
+            project = None
         return (
+            project and
```
Comment (marcua) [inline orchestra/todos/auth.py]: do you need `project and`, and if you do, should there be parenthesis around the next two boolean checks?

File: orchestra/todos/views.py [hunk 12]
```diff
@@ -37,13 +40,34 @@ def update_todos_from_todolist_template(request):
...
+
+
 class TodoList(generics.ListCreateAPIView):
     permission_classes = (permissions.IsAuthenticated,
                           IsAssociatedWithProject)
 
-    serializer_class = TodoSerializer
     queryset = Todo.objects.all()
 
+    def get_serializer_class(self):
```
Comment (marcua) [inline orchestra/todos/views.py:87]: why this change to Todo?

File: orchestra/admin.py [hunk 13]
```diff
@@ -25,6 +25,7 @@
 from orchestra.models import TaskAssignment
```
Comment (marcua) [inline orchestra/admin.py:25]: high-level: i don't quite understand the difference between `todoqa` and `recommendation`. I get why we need `todoqa`, but why do we need `recommendation`. Can we eliminate the second one?

File: orchestra/admin.py [hunk 14]
```diff
@@ -25,6 +25,7 @@
...
 from orchestra.models import TimeEntry
 from orchestra.models import Todo
```
Comment (marcua) [inline orchestra/admin.py:27]: high-level: i had imagined the QA interface would be some admin interface or some other screen in Orchestra. I don't quite get the flow---who loads which screen to do QA, and where does it end up showing up? do we have to add a QA screen to each task that we want it on now? i'd prefer to avoid that.

maybe it just shows up below checklists if you're an admin or something?

File: orchestra/admin.py (hunk 13 above)
Comment (marcua) [reply orchestra/admin.py:25]: is the idea that recommendations show up in subsequent tasks? if so, send a loom video of that experience.

File: orchestra/todos/views.py [hunk 15]
```diff
@@ -86,6 +110,30 @@ def perform_update(self, serializer):
...
 
+class TodoQADetail(generics.RetrieveUpdateAPIView):
+    permission_classes = (permissions.IsAuthenticated,
+                          IsAssociatedWithTodosProject)
+
+    serializer_class = TodoQASerializer
+    queryset = TodoQA.objects.all()
+
+
+class TodoQAList(generics.ListCreateAPIView):
```
Comment (adbharadwaj) [inline orchestra/todos/views.py]: Not required anymore.

File: orchestra/todos/views.py [hunk 16]
```diff
@@ -37,13 +40,34 @@ def update_todos_from_todolist_template(request):
...
+
 class TodoList(generics.ListCreateAPIView):
     permission_classes = (permissions.IsAuthenticated,
                           IsAssociatedWithProject)
 
-    serializer_class = TodoSerializer
     queryset = Todo.objects.all()
 
+    def get_serializer_class(self):
+        if self.request.user.groups.filter(
```
Comment (adbharadwaj) [inline orchestra/todos/views.py]: Add comment.

File: orchestra/models/core/models.py (hunk 7 above)
Comment (adbharadwaj) [reply orchestra/models/core/models.py:663]: I don't think so. At this point the reviewer will be the admin user.

File: orchestra/models/core/models.py (hunk 3 above)
Comment (adbharadwaj) [reply orchestra/models/core/models.py]: This was intentional. If I were to use `approved_datetime` there will be only two states - approved or disapproved. But if I use `NullBooleanField`, I can store three states - approved, disapproved and approval pending (Null).

File: orchestra/static/orchestra/todos/todo-checklist.html (hunk 6 above)
Comment (adbharadwaj) [reply orchestra/static/orchestra/todos/todo-checklist.html]: We cannot because the primary use case is to show the recommendation on the next project where todo ids will not match.

File: orchestra/migrations/0078_todoqa.py (hunk 2 above)
Comment (adbharadwaj) [reply orchestra/migrations/0078_todoqa.py:1]: done.

File: orchestra/admin.py (hunk 13 above)
Comment (adbharadwaj) [reply orchestra/admin.py:25]: sending you the loom videos for the end to end experience.

File: orchestra/admin.py (hunk 14 above)
Comment (adbharadwaj) [reply orchestra/admin.py]: It shows up in the customer success managers launch boost task.

Comment (adbharadwaj) [reply orchestra/admin.py:27]: It shows up on the task page only if you are a project admin.

File: orchestra/admin.py [hunk 17]
```diff
@@ -25,6 +25,7 @@
...
+from orchestra.models import TodoQA
 from orchestra.models import TodoListTemplate
```
Comment (marcua) [inline orchestra/admin.py:29]: We should indicate "feedback on this project" and "feedback from previous project"

File: orchestra/admin.py [hunk 18]
```diff
@@ -25,6 +25,7 @@
...
 from orchestra.models import Worker
```
Comment (marcua) [inline orchestra/admin.py:30]: We should auto-expand any todo that has feedback on it to increase discoverability

File: orchestra/static/orchestra/qa/qa.directive.es6.js [hunk 19]
```diff
@@ -0,0 +1,67 @@
+import { defaults } from 'lodash'
```
Comment (marcua) [inline orchestra/static/orchestra/qa/qa.directive.es6.js]: todos/todos-qa.directive.es6.js

if you feel strongly, it can live in `todos-qa/`

File: orchestra/admin.py [hunk 20]
```diff
@@ -25,6 +25,7 @@
...
 from orchestra.models import WorkerCertification
```
Comment (marcua) [inline orchestra/admin.py:31]: should we explicitly show approvals so the expert knows we gave positive feedback?

File: orchestra/static/orchestra/qa/qa.html [hunk 21]
```diff
@@ -0,0 +1,24 @@
+<section class="section-panel todo-list">
```
Comment (marcua) [inline orchestra/static/orchestra/qa/qa.html]: same thing w/ filename and directory

File: orchestra/static/orchestra/qa/qa.module.es6.js [hunk 22]
```diff
@@ -0,0 +1,17 @@
+/* global angular */
```
Comment (marcua) [inline orchestra/static/orchestra/qa/qa.module.es6.js]: same thing w/ filename

File: orchestra/static/orchestra/todos/todo-qa-list.html [hunk 23]
```diff
@@ -0,0 +1,45 @@
+<script type="text/ng-template" id="todoQaItemTemplate">
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-qa-list.html:1]: high-level: should the qa.directive.* be merged with this so there are less components? unlike React, there's not too much benefit to the multiple nested component approach

File: orchestra/todos/auth.py [hunk 24]
```diff
@@ -35,8 +43,15 @@ def has_permission(self, request, view):
...
             # Create calls have a task ID
             task_id = request.data.get('task')
+            todo_id = request.data.get('todo')
             try:
-                task = Task.objects.get(id=task_id) if task_id else None
+                if task_id:
+                    task = Task.objects.get(id=task_id)
+                elif todo_id:
+                    todo = Todo.objects.get(id=todo_id)
+                    task = Task.objects.get(id=todo.task.id) if todo else None
```
Comment (marcua) [inline orchestra/todos/auth.py]: just: `task = todo.task`
* No need for `if todo` since that would fail on the previous line
* No need for the extra `Todo` query since it's already on `todo.task`

File: orchestra/todos/views.py [hunk 25]
```diff
@@ -37,13 +40,35 @@ def update_todos_from_todolist_template(request):
         raise BadRequest('TodoList Template not found for the given slug.')
 
 
+@api_endpoint(methods=['GET'],
+              permissions=(permissions.IsAuthenticated,
+                           IsAssociatedWithProject),
+              logger=logger)
+def recommendations(request):
```
Comment (marcua) [inline orchestra/todos/views.py]: recommendations is a loaded term for us---call them previous_todo_qas?

File: orchestra/todos/views.py [hunk 26]
```diff
@@ -37,13 +40,35 @@ def update_todos_from_todolist_template(request):
         raise BadRequest('TodoList Template not found for the given slug.')
 
 
+@api_endpoint(methods=['GET'],
```
Comment (marcua) [inline orchestra/todos/views.py:44]: high-level: When do you stop showing feedback on a particular todo? Maybe the most recent N tasks?

File: orchestra/todos/views.py [hunk 27]
```diff
@@ -37,13 +40,35 @@ def update_todos_from_todolist_template(request):
...
+              permissions=(permissions.IsAuthenticated,
+                           IsAssociatedWithProject),
+              logger=logger)
+def recommendations(request):
+    try:
+        todoQAs = TodoQA.objects.filter(
+            todo__task__assignments__worker__user=request.user, approved=False)
+        todos_recommendation = {todoQA.todo.description: TodoQASerializer(
```
Comment (marcua) [inline orchestra/todos/views.py]: are we just basing things on overlapping descriptions for now? OK to say yes---just sanity-checking :)

File: orchestra/static/orchestra/qa/qa.directive.es6.js (hunk 19 above)
Comment (adbharadwaj) [reply orchestra/static/orchestra/qa/qa.directive.es6.js]: I thought about it. If we do, we should also move the backend code (TodoQA) to a different repo.

File: orchestra/todos/views.py (hunk 25 above)
Comment (adbharadwaj) [reply orchestra/todos/views.py]: Ok.

File: orchestra/todos/views.py (hunk 27 above)
Comment (adbharadwaj) [reply orchestra/todos/views.py]: Yes. we are basing things on overlapping descriptions.

File: orchestra/admin.py (hunk 20 above)
Comment (adbharadwaj) [reply orchestra/admin.py:31]: Approvals might be a good idea when giving "feedback on this project" but not so much when giving "feedback from the previous project". If we show them a lot of comments the mistakes and their related comments might get lost in the information pool.

File: orchestra/admin.py (hunk 17 above)
Comment (adbharadwaj) [reply orchestra/admin.py:29]: Done. Here is how the feedback will look like:
screen shot 2018-07-11 at 5 13 04 pm

File: orchestra/todos/views.py (hunk 26 above)
Comment (adbharadwaj) [reply orchestra/todos/views.py:44]: We stop showing feedback if it was created more than 90 days ago.

File: orchestra/todos/views.py (hunk 12 above)
Comment (adbharadwaj) [reply orchestra/todos/views.py:87]: Only include todo QA data for users in the `project_admins` group.

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 28]
```diff
@@ -28,14 +28,21 @@
...
+        <a href="#" ng-hide="!showSkipped" ng-click="unskipTodo(todo)">[relevant]</a>
+        <a href="#" ng-hide="showSkipped" ng-click="skipTodo(todo)">[not relevant]</a>
         <a href="#" ng-hide="todo.parent_todo" ng-click="removeTodo(todo)">[remove]</a>
       </span>
     </label>
+    <div class="todo-qa panel panel-default" ng-if="todoQas[todo.description] && todoQas[todo.description].comment">
+      <div class="panel-body">
+        <strong ng-if="todoQas[todo.description].todo === todo.id">Feedback on this project</strong>
+        <strong ng-if="todoQas[todo.description].todo !== todo.id">Feedback from previous project</strong>
+        : {{todoQas[todo.description].comment}}
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-checklist.html]: no space before colon :)

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 29]
```diff
@@ -28,14 +28,21 @@
...
-        <a href="#" ng-hide="!showSkipped" ng-click="unskipTodo(todo)">[add]</a>
-        <a href="#" ng-hide="showSkipped" ng-click="skipTodo(todo)">[skip]</a>
+        <a href="#" ng-hide="!showSkipped" ng-click="unskipTodo(todo)">[relevant]</a>
+        <a href="#" ng-hide="showSkipped" ng-click="skipTodo(todo)">[not relevant]</a>
         <a href="#" ng-hide="todo.parent_todo" ng-click="removeTodo(todo)">[remove]</a>
       </span>
     </label>
+    <div class="todo-qa panel panel-default" ng-if="todoQas[todo.description] && todoQas[todo.description].comment">
+      <div class="panel-body">
+        <strong ng-if="todoQas[todo.description].todo === todo.id">Feedback on this project</strong>
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-checklist.html]: I wouldn't make these strong---the emphasis is on the feedback

**Review by marcua — APPROVED** [review]

File: orchestra/todos/views.py [hunk 30]
```diff
@@ -37,13 +40,35 @@ def update_todos_from_todolist_template(request):
         raise BadRequest('TodoList Template not found for the given slug.')
 
 
+@api_endpoint(methods=['GET'],
+              permissions=(permissions.IsAuthenticated,
+                           IsAssociatedWithProject),
+              logger=logger)
+def recommendations(request):
+    try:
+        todoQAs = TodoQA.objects.filter(
```
Comment (marcua) [inline orchestra/todos/views.py]: an older one might clobber the current task's feedback

File: orchestra/todos/views.py [hunk 31]
```diff
@@ -37,13 +41,41 @@ def update_todos_from_todolist_template(request):
...
+              permissions=(permissions.IsAuthenticated,
+                           IsAssociatedWithProject),
+              logger=logger)
+def worker_recent_todo_qas(request):
+    """
+    The function returns TodoQAs from the requesting user's most recent
+    task assignment provided they were created in the last 90 days.
+    """
+    try:
+        todoQAs = TodoQA.objects.filter(
```
Comment (marcua) [inline orchestra/todos/views.py]: i feel like for this description clobbering to work, you have to order by (is this the same task, older tasks)

**Review by marcua — CHANGES_REQUESTED** [review]
three small things

File: orchestra/admin.py (hunk 18 above)
Comment (adbharadwaj) [reply orchestra/admin.py:30]: done.

File: orchestra/todos/views.py (hunk 31 above)
Comment (adbharadwaj) [reply orchestra/todos/views.py]: Fixed it and updated the tests to make sure that we are only loading todo qas from the most recent task for the given worker.

File: orchestra/todos/views.py [hunk 32]
```diff
@@ -48,15 +47,17 @@ def update_todos_from_todolist_template(request):
...
-            approved=False,
-            created_at__gte=datetime.now() - timedelta(days=90))
-        todos_recommendation = {todoQA.todo.description: TodoQASerializer(
-            todoQA).data for todoQA in todoQAs}
+        most_recent_worker_task = TodoQA.objects.filter(
+            todo__task__assignments__worker__user=request.user
+        ).order_by('-created_at').first().todo.task
+        todo_qas = TodoQA.objects.filter(
+            todo__task=most_recent_worker_task,
+            approved=False).order_by('-created_at')
```
Comment (marcua) [inline orchestra/todos/views.py]: no need to order these right?

File: orchestra/todos/views.py [hunk 33]
```diff
@@ -48,15 +47,17 @@ def update_todos_from_todolist_template(request):
...
     try:
-        todoQAs = TodoQA.objects.filter(
-            todo__task__assignments__worker__user=request.user,
-            approved=False,
-            created_at__gte=datetime.now() - timedelta(days=90))
-        todos_recommendation = {todoQA.todo.description: TodoQASerializer(
-            todoQA).data for todoQA in todoQAs}
+        most_recent_worker_task = TodoQA.objects.filter(
+            todo__task__assignments__worker__user=request.user
+        ).order_by('-created_at').first().todo.task
```
Comment (marcua) [inline orchestra/todos/views.py]: .first() might be `None` so test for that and only add to `todos_recommendation` if it exists

File: orchestra/todos/views.py [hunk 34]
```diff
@@ -48,15 +47,17 @@ def update_todos_from_todolist_template(request):
...
     """
     try:
-        todoQAs = TodoQA.objects.filter(
-            todo__task__assignments__worker__user=request.user,
-            approved=False,
-            created_at__gte=datetime.now() - timedelta(days=90))
-        todos_recommendation = {todoQA.todo.description: TodoQASerializer(
-            todoQA).data for todoQA in todoQAs}
+        most_recent_worker_task = TodoQA.objects.filter(
+            todo__task__assignments__worker__user=request.user
```
Comment (marcua) [inline orchestra/todos/views.py]: may well put `approved=False` here

File: orchestra/todos/views.py [hunk 35]
```diff
@@ -48,15 +47,17 @@ def update_todos_from_todolist_template(request):
...
+        ).order_by('-created_at').first().todo.task
+        todo_qas = TodoQA.objects.filter(
+            todo__task=most_recent_worker_task,
+            approved=False).order_by('-created_at')
+        todos_recommendation = {todo_qa.todo.description: TodoQASerializer(
+            todo_qa).data for todo_qa in todo_qas}
     except TodoQA.DoesNotExist:
```
Comment (marcua) [inline orchestra/todos/views.py]: I don't think this exception can happen

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 36]
```diff
@@ -35,14 +35,13 @@
     </label>
     <div class="todo-qa panel panel-default" ng-if="todoQas[todo.description] && todoQas[todo.description].comment">
       <div class="panel-body">
-        <strong ng-if="todoQas[todo.description].todo === todo.id">Feedback on this project</strong>
-        <strong ng-if="todoQas[todo.description].todo !== todo.id">Feedback from previous project</strong>
-        : {{todoQas[todo.description].comment}}
+        <span>{{todoQas[todo.description].todo === todo.id?"Feedback on this project":"Feedback from previous project"}}:</span>
+        <strong>{{todoQas[todo.description].comment}}</strong>
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-checklist.html]: sanity-check: is there a space after the colon?

File: orchestra/todos/views.py (hunk 34 above)
Comment (adbharadwaj) [reply orchestra/todos/views.py]: That may be wrong in a scenario where the worker has done everything correctly (all approved). It would seem like reporting mistakes even though they did perfectly last time. :)

File: orchestra/todos/views.py (hunk 32 above)
Comment (adbharadwaj) [reply orchestra/todos/views.py]: yes

File: orchestra/static/orchestra/todos/todo-checklist.html (hunk 36 above)
Comment (adbharadwaj) [reply orchestra/static/orchestra/todos/todo-checklist.html]: Yes.

Comment (adbharadwaj) [reply orchestra/static/orchestra/todos/todo-checklist.html]: Here is how it looks now.
screen shot 2018-07-12 at 10 26 58 am

**Review by marcua — APPROVED** [review]
